tweaks-qt/maindialog.cpp

```diff
@@ -36,13 +43,20 @@ MainDialog::~MainDialog()
     xml_finish();
 }
 
-static const char *first_field(char *s, char delim)
+void MainDialog::addSelectedLayout(void)
 {
-    char *p = strchr(s, delim);
-    if (p) {
-        *p = '\0';
+    const char *description = ui->layoutCombo->currentText().toLatin1().data();
+    for (int i = 0; i < ARRAY_SIZE(evdev_lst_layouts); ++i) {
```

Comment:
replace `evdev_lst_layout` with a `std::map` (or `QHash`) and use c++11
`for (auto& layout : evdev_lst_layouts)` (not using ARRAY_SIZE, nor Qt foreach)?

Reply:
Thanks. Good idea. I went for a `std::vector` to make the code a bit easier to read with variable names `code` and `description`.